### PR TITLE
solve cicd issue

### PR DESCRIPTION
**Pull request by houwenchen**
## Pull Request template

Please, go through these steps before you submit a PR.

**Why is this PR required? What issue does it fix?**:

**What this PR does?**:

**Does this PR require any upgrade changes?**:

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Any additional information for your reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

**PLEASE REMOVE BELOW INFORMATION BEFORE SUBMITTING**

The PR title message must follow convention:
   `<type>(<scope>): <subject>`.

Where: <br />
- `type` is defining if release will be triggering after merging submitted changes, details in [CONTRIBUTING.md](../CONTRIBUTING.md).
    Most common types are:
    - `feat`      - for new features, not a new feature for build script
    - `fix`       - for bug fixes or improvements, not a fix for build script
    - `chore`     - changes not related to production code
    - `docs`      - changes related to documentation
    - `style`     - formatting, missing semi colons, linting fix etc; no significant production code changes
    - `test`      - adding missing tests, refactoring tests; no production code change
    - `refactor`  - refactoring production code, eg. renaming a variable or function name, there should not be any significant production code changes

- `scope` is a single word that best describes where the changes fit.
  Most common scopes are like:
    - data engine (`localpv`, `jiva`, `cstor`)
    - feature (`provisioning`, `backup`, `restore`, `exporter`)
    - code component (`api`, `webhook`, `cast`, `upgrade`)
    - test (`tests`, `bdd`)
    - chores (`version`, `build`, `log`, `travis`)

- `subject` is a single line brief description of the changes made in the pull request.

### CODE REVIEW

**Review by Abhinandan-Purkait — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -60,11 +60,22 @@ jobs:
...
+          cache: false
+
       - name: Unit test
         run: make test
 
       - name: Upload Coverage Report
-        uses: codecov/codecov-action@v1
+        uses: codecov/codecov-action@v3  
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```
Comment (niladrih) [inline .github/workflows/build.yml:75]: This token does not exist for this repo. Is this required?

**Review by niladrih — COMMENTED** [review]
Left a comment

File: .github/workflows/build.yml (hunk 1 above)
Comment (houwenchen) [reply .github/workflows/build.yml:75]: yes, should add a codecov token in this project.

File: Makefile [hunk 2]
```diff
@@ -110,7 +110,7 @@ format:
 .PHONY: test
 test: format
 	@echo "--> Running go test" ;
-	@go test $(UNIT_TEST_PACKAGES)
+	@go test $(UNIT_TEST_PACKAGES) -cover  -coverpkg=./... -covermode=atomic -coverprofile=coverage.txt
```
Comment (niladrih) [inline Makefile:113]: This is already present in /buildscripts/test-cov.sh

**Comment by niladrih** [comment]
I think this change should use the script `/buildscripts/test-cov.sh` in the 'test' Makefile target, and then try to publish the codecov result without a token? Token is usually required when there are chances of a large number of github api calls, I think.

**Comment by houwenchen** [comment]
> I think this change should use the script `/buildscripts/test-cov.sh` in the 'test' Makefile target, and then try to publish the codecov result without a token? Token is usually required when there are chances of a large number of github api calls, I think.

**Comment by houwenchen** [comment]
@niladrih , it's just like docker token to upload docker image.

**Review by niladrih — COMMENTED** [review]
@houwenchen -- these changes also need to go into pull_request.yml

**Comment by Abhinandan-Purkait** [comment]
Has been cherry-picked.

**Pull request closed without merging**